src/Container.php

```diff
@@ -92,7 +92,7 @@ public function get($resourceName)
             throw new KeyNotFoundException(sprintf("Resource '%s' has not been registered with the container.", $resourceName));
         }
 
-        return $this->resources[$key]->getInstance();
+        return $this->resources[$key]->getInstance(false);
```

Comment:
Does this have to be false?

Reply:
The `noProxy` is only needed for resource extending.  I think we can get rid of it if we move the `extend` method to `ContainerResource` class.
Then `Container:extend()` method would look like this:

```pho
public function extend($key, $callable)
{
    $container->resources[$key]->extend($callable);

    return $this;
}
```

Reply:
I've created a separate PR implementing this idea: https://github.com/joomla-framework/di/pull/62

It allows us not to change `getInstance()` at all.

The whole changes you can see in another branch: https://github.com/joomla-framework/di/compare/3.x-dev...PopArtDesign:di:lazy-services-extending